fastlane/lanes/release.rb

```diff
@@ -118,31 +211,78 @@
+
+    # Parse the provided version into an AppVersion object
+    parsed_version = VERSION_FORMATTER.parse(new_version)
+    previous_version = VERSION_FORMATTER.release_version(VERSION_CALCULATOR.previous_patch_version(version: parsed_version))
+
+    # Check versions
+    message = <<-MESSAGE
+      New Hotfix:
+
+      • Current release version and build code: #{release_version_current} (#{build_code_current}).
+      • New release version and build code: #{new_version} (#{build_code_hotfix}).
+
+      • Current internal release version and build code: #{release_version_current_internal} (#{build_code_current_internal}).
+      • New internal release version and build code: #{new_version} (#{build_code_hotfix_internal(release_version: new_version)}).
+
+      Branching from tag: #{previous_version}
+    MESSAGE
+
+    UI.important(message)
+    UI.user_error!('Aborted by user request') if !options[:skip_confirm] && !UI.confirm('Do you want to continue?')
+
+    # Check tags
+    UI.user_error!("Version #{new_version} already exists! Abort!") if git_tag_exists(tag: new_version)
+    UI.user_error!("Version #{previous_version} is not tagged! A hotfix branch cannot be created.") unless git_tag_exists(tag: previous_version)
+
+    # Create the hotfix branch
+    UI.message 'Creating hotfix branch...'
+    # Verify that there's nothing in progress in the working copy
+    ensure_git_status_clean
+    Fastlane::Helper::GitHelper.create_branch("release/#{new_version}", from: previous_version)
+    UI.success "Done! New hotfix branch is: #{git_branch}"
+
+    # Bump the hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping hotfix version and build code...'
+    PUBLIC_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: build_code_hotfix
+    )
+    UI.success "Done! New Release Version: #{release_version_current}. New Build Code: #{build_code_current}"
+
+    # Bump the internal hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping internal hotfix version and build code...'
+    INTERNAL_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: build_code_hotfix_internal
     )
+    UI.success "Done! New Internal Release Version: #{release_version_current_internal}. New Internal Build Code: #{build_code_current_internal}"
+
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
   end
 
   # Finalizes a hotfix, by triggering a release build on CI
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
   #
   desc 'Performs the final checks and triggers a release build for the hotfix in the current branch'
-  lane :finalize_hotfix_release do |options|
-    ios_finalize_prechecks(options)
+  lane :finalize_hotfix_release do
+    # Verify that the current branch is a release branch. Notice that `ensure_git_branch` expects a RegEx parameter
+    ensure_git_branch(branch: '^release/')
```

Comment:
I wonder if we should improve this check to ensure it has a `x.y.z` version number in the branch name, to make sure it's a hotfix.

Reply:
Good idea. That will be an easy method to add to the [`abstract_version_calculator.rb`](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/versioning/calculators/abstract_version_calculator.rb) versioning class of the Release Toolkit. I will make a note to add a method there that we can use.

Reply:
> I will make a note to add a `hotfix?` method there that can we use for lanes like this.

Nice.